# Importing necessary libraries

In [1]:
# Import necessary libraries
import pandas as pd
import numpy as np
import json
import os
from pathlib import Path
import warnings
warnings.filterwarnings('ignore')

from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import (accuracy_score, precision_score, recall_score, 
                             f1_score, roc_auc_score, confusion_matrix, 
                             classification_report, roc_curve)
import matplotlib.pyplot as plt
import seaborn as sns


# Data loading and exploration

## Define the merge function


The following code block will:
- Find all CSV files in the data directory
- Read each CSV file into a pandas DataFrame
- Combine all DataFrames into one


In [2]:

# Get the path to the data directory using Path for cross-platform compatibility
data_path = Path('data')

# Find all CSV files in the directory and sort them for consistent ordering
# glob('*.csv') finds all files ending in .csv
csv_files = sorted(data_path.glob('*.csv'))

if not csv_files:
    raise ValueError(f"No CSV files found in {'data'}")

print(f"Found {len(csv_files)} CSV files to merge...")

# Read all CSV files into a list of DataFrames
# We'll store each DataFrame in a list before concatenating
dataframes = []
for csv_file in csv_files:
    print(f"Reading {csv_file.name}...")
    try:
        # pd.read_csv() reads the CSV file and converts it to a DataFrame
        df = pd.read_csv(csv_file)
        dataframes.append(df)
    except Exception as e:
        # If a file can't be read, print a warning but continue with other files
        print(f"Warning: Could not read {csv_file.name}: {e}")
        continue

if not dataframes:
    raise ValueError("No dataframes were successfully loaded")

# Concatenate all DataFrames into one
# ignore_index=True creates a new sequential index (0, 1, 2, ...) instead of keeping original indices
print("Merging dataframes...")
merged_df = pd.concat(dataframes, ignore_index=True)

# Display summary information about the merged DataFrame
print(f"\nMerged DataFrame shape: {merged_df.shape}")
print(f"Total rows: {len(merged_df)}")
print(f"Total columns: {len(merged_df.columns)}")


Found 83 CSV files to merge...
Reading Kickstarter.csv...
Reading Kickstarter001.csv...
Reading Kickstarter002.csv...
Reading Kickstarter003.csv...
Reading Kickstarter004.csv...
Reading Kickstarter005.csv...
Reading Kickstarter006.csv...
Reading Kickstarter007.csv...
Reading Kickstarter008.csv...
Reading Kickstarter009.csv...
Reading Kickstarter010.csv...
Reading Kickstarter011.csv...
Reading Kickstarter012.csv...
Reading Kickstarter013.csv...
Reading Kickstarter014.csv...
Reading Kickstarter015.csv...
Reading Kickstarter016.csv...
Reading Kickstarter017.csv...
Reading Kickstarter018.csv...
Reading Kickstarter019.csv...
Reading Kickstarter020.csv...
Reading Kickstarter021.csv...
Reading Kickstarter022.csv...
Reading Kickstarter023.csv...
Reading Kickstarter024.csv...
Reading Kickstarter025.csv...
Reading Kickstarter026.csv...
Reading Kickstarter027.csv...
Reading Kickstarter028.csv...
Reading Kickstarter029.csv...
Reading Kickstarter030.csv...
Reading Kickstarter031.csv...
Reading Kick

## Explore the merged data

Let's take a look at the structure and content of our merged DataFrame.


In [3]:
# Display the first few rows to see what the data looks like
# head() shows the first 5 rows by default
merged_df.head()


,backers_count,blurb,category,converted_pledged_amount,country,country_displayable_name,created_at,creator,currency,currency_symbol,...,spotlight,staff_pick,state,state_changed_at,static_usd_rate,urls,usd_exchange_rate,usd_pledged,usd_type,video
0,20,A photographic book of a festival celebrated b...,"{""id"":278,""name"":""People"",""analytics_name"":""Pe...",637.0,GB,the United Kingdom,1724064660,"{""id"":729000581,""name"":""Neil Martinson"",""slug""...",GBP,£,...,False,False,failed,1727098184,1.320305,"{""web"":{""project"":""https://www.kickstarter.com...",1.330982,632.425889,domestic,"{""id"":1312771,""status"":""successful"",""hls"":""htt..."
1,531,Unlock the Full Potential of Your Steam Deck w...,"{""id"":52,""name"":""Hardware"",""analytics_name"":""H...",83872.0,US,the United States,1746583126,"{""id"":767329947,""name"":""Seesaw"",""slug"":""dockca...",USD,$,...,False,True,live,1761051614,1.000000,"{""web"":{""project"":""https://www.kickstarter.com...",1.000000,83872.000000,domestic,"{""id"":1367591,""status"":""successful"",""hls"":""htt..."
2,0,Retagger Mobile: Transform the Internet with S...,"{""id"":332,""name"":""Apps"",""analytics_name"":""Apps...",0.0,US,the United States,1731435583,"{""id"":1696039535,""name"":""Donnie Bugden"",""slug""...",USD,$,...,False,False,live,1760906735,1.000000,"{""web"":{""project"":""https://www.kickstarter.com...",1.000000,0.000000,domestic,"{""id"":1392212,""status"":""successful"",""hls"":""htt..."
3,166,Help support the making and release of Armchai...,"{""id"":37,""name"":""Country & Folk"",""analytics_na...",12314.0,US,the United States,1548701666,"{""id"":1417057582,""name"":""Armchair Boogie"",""slu...",USD,$,...,True,True,successful,1559659892,1.000000,"{""web"":{""project"":""https://www.kickstarter.com...",1.000000,12314.860000,domestic,"{""id"":955599,""status"":""successful"",""hls"":""http..."
4,66,When you buy MatchBack's premier graphic tee &...,"{""id"":263,""name"":""Apparel"",""analytics_name"":""A...",5599.0,US,the United States,1496003666,"{""id"":2025826474,""name"":""Aaron Podell"",""is_reg...",USD,$,...,True,False,successful,1498892340,1.000000,"{""web"":{""project"":""https://www.kickstarter.com...",1.000000,5599.000000,domestic,"{""id"":787581,""status"":""successful"",""hls"":null,..."


In [4]:
# Get basic information about the DataFrame
# info() shows column names, data types, and non-null counts
merged_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 265019 entries, 0 to 265018
Data columns (total 42 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   backers_count                       265019 non-null  int64  
 1   blurb                               264909 non-null  object 
 2   category                            265019 non-null  object 
 3   converted_pledged_amount            248444 non-null  float64
 4   country                             265019 non-null  object 
 5   country_displayable_name            265019 non-null  object 
 6   created_at                          265019 non-null  int64  
 7   creator                             265019 non-null  object 
 8   currency                            265019 non-null  object 
 9   currency_symbol                     265019 non-null  object 
 10  currency_trailing_code              265019 non-null  bool   
 11  current_currency          

In [5]:
# Display basic statistics for numeric columns
# describe() provides count, mean, std, min, max, and quartiles
merged_df.describe()


,backers_count,converted_pledged_amount,created_at,deadline,fx_rate,goal,id,launched_at,percent_funded,pledged,state_changed_at,static_usd_rate,usd_exchange_rate,usd_pledged
count,265019.000000,2.484440e+05,2.650190e+05,2.650190e+05,265019.000000,2.650190e+05,2.650190e+05,2.650190e+05,2.650190e+05,2.650190e+05,2.650190e+05,265019.000000,248444.000000,2.484440e+05
mean,133.218079,1.855043e+04,1.596649e+09,1.504914e+09,0.975356,5.333726e+04,1.074539e+09,1.494149e+09,7.286407e+02,5.463605e+04,1.604202e+09,0.918595,0.979627,1.855117e+04
std,723.039697,2.237439e+05,1.342557e+08,3.951988e+08,0.266555,2.939503e+06,6.191940e+08,4.074183e+08,5.444585e+04,1.073879e+06,1.343778e+08,0.351749,0.267911,2.236557e+05
min,0.000000,0.000000e+00,1.240366e+09,0.000000e+00,0.005584,0.000000e+00,1.358300e+04,0.000000e+00,0.000000e+00,0.000000e+00,1.242468e+09,0.000000,0.006186,0.000000e+00
25%,3.000000,2.410000e+02,1.475000e+09,1.452541e+09,1.000000,1.000000e+03,5.385046e+08,1.447216e+09,2.066607e+00,1.120000e+02,1.481911e+09,1.000000,1.000000,2.416224e+02
50%,26.000000,2.132000e+03,1.623565e+09,1.595955e+09,1.000000,4.500000e+03,1.074939e+09,1.589217e+09,1.028500e+02,1.856000e+03,1.633094e+09,1.000000,1.000000,2.134045e+03
75%,89.000000,8.277000e+03,1.724666e+09,1.726081e+09,1.000000,1.200000e+04,1.610111e+09,1.722500e+09,1.457500e+02,8.390000e+03,1.732712e+09,1.000000,1.000000,8.273000e+03
max,105857.000000,4.676226e+07,1.762918e+09,1.768118e+09,1.313030,1.000000e+09,2.147476e+09,1.762956e+09,1.553204e+07,1.755535e+08,1.762956e+09,1.716408,1.716408,4.676226e+07


# Classification Model

This section of the notebook builds a classification model to predict whether a Kickstarter project will be "successful" or "failed" using only features available at project launch.


## Techniques and Algorithms Used


1. Random Forest Classifier
   - Bagged ensemble of decision trees
   - Each tree uses random sample with replacement
   - Final prediction is mean score across trees

2. GridSearchCV for Hyperparameter Tuning
   - Examines all possible combinations of specified hyperparameters
   - Uses cross-validation to find optimal parameters

3. 5-Fold Cross-Validation
   - Splits training data into k parts
   - Each fold serves as validation data once

4. One-Hot Encoding
   - Encodes categorical variables with boolean variables (0/1)

5. Classification Metrics
   - Confusion Matrix, Accuracy, Precision, Recall, F1, ROC-AUC

6. Feature Importance
   - Understanding which features are most impactful

7. Train-Test Split
   - 80/20 split with stratification

## Feature Engineering

Only features available at project LAUNCH are used. This ensures the model can make predictions when a project is launched, not after it has already received funding.

Excluded FEATURES (NOT available at launch):
- `backers_count`: Number of backers (only known after launch)
- `pledged`: Amount pledged (only known after launch)
- `staff_pick`: Whether Kickstarter staff featured the project (selected after launch)
- `percent_funded`: Funding percentage (only known after launch)
- `usd_pledged`: USD pledged (only known after launch)
- `spotlight`: Spotlight status (assigned after campaign ends)
- `state_changed_at`: When state changed (after campaign ends)

### Filter Data by Project State


We only include projects with state = "successful" or "failed" for our binary classification model. Projects with other states (e.g., "live", "canceled", "suspended") are excluded as they represent incomplete or ambiguous outcomes.


In [6]:
# Filter to only include projects with state = "successful" or "failed"
# Create a copy of merged_df for modeling
df = merged_df.copy()

# Display state distribution before filtering
print("Project states before filtering:")
print(df['state'].value_counts())
print(f"\nTotal projects before filtering: {len(df)}")

# Filter to only successful and failed projects
df = df[df['state'].isin(['successful', 'failed'])].copy()

# Display state distribution after filtering
print("\nProject states after filtering:")
print(df['state'].value_counts())
print(f"\nTotal projects after filtering: {len(df)}")
print(f"Projects removed: {len(merged_df) - len(df)}")


Project states before filtering:
state
successful    152296
failed         77302
submitted      13782
live            9494
canceled        9352
started         2783
suspended         10
Name: count, dtype: int64

Total projects before filtering: 265019

Project states after filtering:
state
successful    152296
failed         77302
Name: count, dtype: int64

Total projects after filtering: 229598
Projects removed: 35421


In [7]:
# Extract parent category from JSON
def extract_parent_category(cat_str):
    """Extract the parent category from the nested JSON category field."""
    try:
        cat_dict = json.loads(cat_str.replace('""', '"'))
        return cat_dict.get('parent_name', 'Unknown')
    except:
        return 'Unknown'

df['parent_category'] = df['category'].apply(extract_parent_category)

In [8]:
# Campaign duration (in days)
# Calculated from deadline and launched_at timestamps
df['campaign_duration_days'] = (df['deadline'] - df['launched_at']) / (60 * 60 * 24)

# Has video (binary: 1 if project has video, 0 otherwise)
df['has_video'] = (~df['video'].isna()).astype(int)

# Blurb length (character count of project description)
df['blurb_length'] = df['blurb'].fillna('').apply(len)

# Name length (character count of project name)
df['name_length'] = df['name'].fillna('').apply(len)

# Goal in USD (standardized funding goal)
df['goal_usd'] = df['goal'] * df['static_usd_rate']

# 7. Country - will be one-hot encoded 

# Log transform of goal (to handle right-skewed distribution)
df['log_goal_usd'] = np.log1p(df['goal_usd'])

# Create target variable (1 = successful, 0 = failed)
df['target'] = (df['state'] == 'successful').astype(int)

print("\nFeatures Created (All available at launch):")
print("-" * 50)
print("1. log_goal_usd:         Log of funding goal in USD")
print("2. campaign_duration_days: Length of campaign in days")
print("3. blurb_length:         Character count of description")
print("4. name_length:          Character count of project name")
print("5. has_video:            Whether project has a video (0/1)")
print("7. country:              Project country (one-hot encoded)")
print("7. parent_category:      Main category (one-hot encoded)")


Features Created (All available at launch):
--------------------------------------------------
1. log_goal_usd:         Log of funding goal in USD
2. campaign_duration_days: Length of campaign in days
3. blurb_length:         Character count of description
4. name_length:          Character count of project name
5. has_video:            Whether project has a video (0/1)
7. country:              Project country (one-hot encoded)
7. parent_category:      Main category (one-hot encoded)


## Prepare Features and Target

In [10]:
# Numerical features
numerical_features = [
    'log_goal_usd',           # Log of funding goal
    'campaign_duration_days', # Campaign length
    'blurb_length',           # Description length
    'name_length',            # Project name length
]

# Binary features
binary_features = [
    'has_video',    # Has video presentation      
]

# Categorical features - will be one-hot encoded
categorical_features = ['parent_category', 'country']

# Prepare the feature matrix
X_num = df[numerical_features].copy()
X_bin = df[binary_features].astype(int).copy()
X_cat = pd.get_dummies(df[categorical_features], drop_first=True)  # drop_first to avoid multicollinearity

# Combine all features
X = pd.concat([X_num, X_bin, X_cat], axis=1)
y = df['target']

print(f"\nFeature Matrix Shape: {X.shape}")
print(f"Number of samples: {X.shape[0]}")
print(f"Number of features: {X.shape[1]}")
print("\nFeature names:")
for i, col in enumerate(X.columns):
    print(f"  {i+1:2d}. {col}")


Feature Matrix Shape: (229598, 44)
Number of samples: 229598
Number of features: 44

Feature names:
   1. log_goal_usd
   2. campaign_duration_days
   3. blurb_length
   4. name_length
   5. has_video
   6. parent_category_Comics
   7. parent_category_Crafts
   8. parent_category_Dance
   9. parent_category_Design
  10. parent_category_Fashion
  11. parent_category_Film & Video
  12. parent_category_Food
  13. parent_category_Games
  14. parent_category_Journalism
  15. parent_category_Music
  16. parent_category_Photography
  17. parent_category_Publishing
  18. parent_category_Technology
  19. parent_category_Theater
  20. parent_category_Unknown
  21. country_AU
  22. country_BE
  23. country_CA
  24. country_CH
  25. country_DE
  26. country_DK
  27. country_ES
  28. country_FR
  29. country_GB
  30. country_GR
  31. country_HK
  32. country_IE
  33. country_IT
  34. country_JP
  35. country_LU
  36. country_MX
  37. country_NL
  38. country_NO
  39. country_NZ
  40. country_PL
  

## Train-Test Split

In [11]:
# Using 80/20 split with stratification to maintain class proportions
X_train, X_test, y_train, y_test = train_test_split(
    X, y, 
    test_size=0.2, 
    random_state=42, 
    stratify=y  # Maintain class balance in both sets
)

print(f"\nTraining set: {X_train.shape[0]} samples ({X_train.shape[0]/len(X)*100:.1f}%)")
print(f"Test set:     {X_test.shape[0]} samples ({X_test.shape[0]/len(X)*100:.1f}%)")
print(f"\nTraining class distribution:")
print(f"  Successful: {(y_train == 1).sum()} ({(y_train == 1).mean()*100:.1f}%)")
print(f"  Failed:     {(y_train == 0).sum()} ({(y_train == 0).mean()*100:.1f}%)")



Training set: 183678 samples (80.0%)
Test set:     45920 samples (20.0%)

Training class distribution:
  Successful: 121837 (66.3%)
  Failed:     61841 (33.7%)


## Baseline Model Comparison

### Logistic Regression

In [ ]:
# Scale features for Logistic Regression
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)


lr = LogisticRegression(random_state=42, max_iter=1000)
lr_cv_scores = cross_val_score(lr, X_train_scaled, y_train, cv=5, scoring='roc_auc')
print(f"1. Logistic Regression:")
print(f"   CV AUC: {lr_cv_scores.mean():.4f} (+/- {lr_cv_scores.std():.4f})")

### Decision Tree

In [ ]:
# Scale features for Logistic Regression
dt = DecisionTreeClassifier(random_state=42, max_depth=5)
dt_cv_scores = cross_val_score(dt, X_train, y_train, cv=5, scoring='roc_auc')
print(f"\n2. Decision Tree (max_depth=5):")
print(f"   CV AUC: {dt_cv_scores.mean():.4f} (+/- {dt_cv_scores.std():.4f})")

### Random Forest

In [ ]:
# Scale features for Logistic Regression
rf = RandomForestClassifier(random_state=42, n_estimators=100)
rf_cv_scores = cross_val_score(rf, X_train, y_train, cv=5, scoring='roc_auc')
print(f"\n3. Random Forest (n_estimators=100):")
print(f"   CV AUC: {rf_cv_scores.mean():.4f} (+/- {rf_cv_scores.std():.4f})")

### Model Comparison

In [ ]:
# Calculate mean CV scores for each model
models = {
    'Logistic Regression': lr_cv_scores.mean(),
    'Decision Tree': dt_cv_scores.mean(),
    'Random Forest': rf_cv_scores.mean()
}

# Display results
for model_name, score in sorted(models.items(), key=lambda x: x[1], reverse=True):
    print(f"{model_name:25s}: {score:.4f}")

# Identify best model
best_model = max(models, key=models.get)
best_score = models[best_model]

print("\n" + "-" * 60)
print(f"BEST MODEL: {best_model}")
print(f"Best CV AUC Score: {best_score:.4f}")
print("=" * 60)


> Random Forest is selected for hyperparameter tuning as it achieved the highest cross-validation ROC-AUC score (0.8599) among all three models. Additionally, Random Forest can capture non-linear relationships in the data through its ensemble of decision trees, making it well-suited for this classification task.

## Hyperparameter Tuning(GridSearchCV)

Using GridSearchCV for hyperparameter optimization

Hyperparameters being tuned:
- `n_estimators`: Number of trees in the forest
- `max_depth`: Maximum depth of each tree
- `min_samples_split`: Minimum samples required to split a node
- `min_samples_leaf`: Minimum samples required at a leaf node

In [ ]:
param_grid = {
    'n_estimators': [100, 150],
    'max_depth': [10, 15, 20],
    'min_samples_split': [2, 5],
    'min_samples_leaf': [1, 2]
}

print(f"Parameter grid: {param_grid}")
print(f"Total combinations to test: {2 * 3 * 2 * 2} = 24")
print(f"With 5-fold CV: {24 * 5} = 120 total fits")

rf = RandomForestClassifier(random_state=42)

grid_search = GridSearchCV(
    estimator=rf,
    param_grid=param_grid,
    cv=5,  # 5-fold cross-validation
    scoring='roc_auc',
    n_jobs=-1,  # Use all available cores
    verbose=1
)

grid_search.fit(X_train, y_train)

print(f"\nBest Parameters Found:")
for param, value in grid_search.best_params_.items():
    print(f"  {param}: {value}")
print(f"\nBest Cross-Validation AUC: {grid_search.best_score_:.4f}")

## Final Model Training and Evaluation

In [ ]:
# Train final model with best parameters
best_rf = grid_search.best_estimator_

# Make predictions on test set
y_pred = best_rf.predict(X_test)
y_pred_proba = best_rf.predict_proba(X_test)[:, 1]

# Confusion Matrix
print("\n1. CONFUSION MATRIX:")
print("-" * 40)
cm = confusion_matrix(y_test, y_pred)
print(f"                    Predicted")
print(f"                  Failed  Successful")
print(f"Actual Failed      {cm[0,0]:4d}      {cm[0,1]:4d}")
print(f"Actual Successful  {cm[1,0]:4d}      {cm[1,1]:4d}")

# Calculate metrics
print("\n2. CLASSIFICATION METRICS:")
print("-" * 40)
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)
roc_auc = roc_auc_score(y_test, y_pred_proba)

print(f"Accuracy:   {accuracy:.4f}  (% of total correctly classified)")
print(f"Precision:  {precision:.4f}  (% of predicted successes that were correct)")
print(f"Recall:     {recall:.4f}  (% of actual successes correctly predicted)")
print(f"F1 Score:   {f1:.4f}  (harmonic mean of precision and recall)")
print(f"ROC-AUC:    {roc_auc:.4f}  (probability ranking performance)")

print("\n3. CLASSIFICATION REPORT:")
print("-" * 40)
print(classification_report(y_test, y_pred, target_names=['Failed', 'Successful']))

## Feature Importance Analysis

In [ ]:
feature_importance = pd.DataFrame({
    'feature': X.columns,
    'importance': best_rf.feature_importances_
}).sort_values('importance', ascending=False)

print("\nFeature Importance Ranking:")
print("-" * 50)
for rank, (_, row) in enumerate(feature_importance.iterrows(), 1):
    print(f"{rank:2d}. {row['feature']:35s} {row['importance']:.4f}")


## Model Interpretation

### Model Logic Explanation

The Random Forest classifier works by building multiple decision trees, each trained on a random subset of the data and features. Each tree makes
a prediction, and the final prediction is the majority vote across all trees. This ensemble approach reduces overfitting and improves generalization.

### Key Findings From Feature Importance

#### 1. Goal Amount `(log_goal_usd)` - MOST IMPORTANT (30.7%)
   - Lower funding goals are more likely to succeed
   - Very high goals make projects harder to fully fund
   - The LOG transformation helps handle the wide range of goal values
     (from hundreds to millions of dollars)
   - Business insight: Creators should set realistic, achievable goals

#### 2. Campaign Duration (`campaign_duration_days`) - SECOND MOST IMPORTANT (20.0%)
   - Campaign length significantly affects funding success
   - Too short campaigns may not allow enough time to build momentum
   - Too long campaigns may lose urgency and backer interest
   - Business insight: Optimal campaign length balances urgency with reach

#### 3. Project Name & Description (`blurb_length` 14.1%, `name_length` 13.7%)
   - Together these account for ~27.8% of predictive power
   - Longer, more descriptive project names attract attention
   - Detailed blurbs provide backers with necessary information
   - Business insight: Invest time in crafting clear, descriptive copy

#### 4. Video Presence (`has_video` 6.9%)
   - Projects with videos are more likely to succeed
   - Videos help communicate vision and build trust with backers
   - Business insight: Creating a compelling video is worth the investment

#### 5. Project Category (`parent_category` - combined ~13.1%)
   - Food projects show the highest category importance (2.3%)
   - Crafts (1.8%) and Journalism (1.6%) categories also show notable effects
   - Most categories have relatively low individual importance (<1%)
   - Different categories have inherently different success dynamics
   - Business insight: Success benchmarks vary by category, but category is less
     important than goal amount, duration, and project description

#### 6. Geographic Location (`is_us` 1.5%)
   - US-based projects show slightly different funding patterns
   - Relatively minor effect compared to other features
   - May reflect payment processing, shipping, or cultural factors


### Actionable Recommendations For Project Creators:

Based on the model's feature importance, creators can improve their
chances of success by:

1. Setting a realistic funding goal (most important factor)
2. Writing a detailed, informative project name and description
3. Choosing an appropriate campaign duration
4. Including a video presentation
5. Understanding their category's typical success patterns